### DISCUSSION TITLE
astyle check for embedded projects

### DISCUSSION THREAD

**Discussion by sathishkumar-24** [SpenceKonde/megaTinyCore#932] [Q&A]
i am trying to do cicd for esp32(espressif) and arm(stm32 make build) projects can any one help thanks
For testing purpose i created one c file and i am trying to use megaTinyCore/.github/workflows/check-code-formatting.yml when run action i am getting my check code formatting was success but my c code was not in formatter following incorrect format code is there
int main(void) {
  uint8_t ch;
int number1,number2,          sum;
   sum = number1+number2; }
![image](https://user-images.githubusercontent.com/109523140/230364626-5ae48cc2-288a-4275-8b5b-9dfc62792b27.png)

**Comment by SpenceKonde** [comment]
If your code was 
```
int main(void) {
uint8_t ch;
int number1,number2, sum;
sum = number1+number2; }
```
Well yeah no wonder that fails the style check! You're violating three style rules right there, all of them ones which I happen to agree with. 
```
int number1,number2, sum;
```
is very poor style and I do not permit code like that in my cores. 
```
int number1, number2, sum;
```
is accepted by astyle only because it doesn't have a way to make that fail too. 
```
int number1;
int number2;
int sum;
```
is what I would require if it were up to me. 
```
sum = number1+number2; 
```
Are you smoking crack or something?!
```
sum = number1 + number2;
```
Finally, your indentation is missing. This is and should be an absolute stylecheck failure:
```
int main(void) {
  uint8_t ch;
  int number1;
  int number2;
  int sum;
  sum = number1 + number2;
}
```
is what I would expect for my core. 

That is also invalid C, so actually I wouldn't merge that either! 
```
int main(void) {
  uint8_t ch; //missing indentation (I often override the indentation check, because astyle has a problem i'll mention below
  int number1; // Each variable declaration on it's own line
  int number2;
  // This is also invalid C because you have declared two local variables but not assigned values to them before you use them. Behavior is undefined and the compiler should warn you 
  int sum;
  sum = number1 + number2; // Space between variable names and operators. 
  return sum; // note that main should never return under normal operation on embedded, but avrgcc will bitch if there isn't some way for it to (though it should be unreachable)
}
```

**Reply by SpenceKonde** [reply]
The problem with astyle is that it doesn't want indentations for preproxcessor directives. That's not acceptable. I *always* indent any conditional of any sort (2 spaces, I have another checker that fails anything with a true tab anywhere outside keywords.txt). 
Because of how much preprocessor stuff I have within the core itself, I don't run AStyle there, but I periodically run a bunch of find in files regexes to catch deviant formatting like:
```
if(conditional){ // missing space between f and (, and between ) and {
```
```
if (conditional)
{ // this violates my style rules for all code. 
``` 
```
uint8_t * var; // inappropriate space. 
```
```
uint8_t* var; // much better
```
```
void (& int paramname) { // inappropriate space
```
```
void (&int paramname) { // much better
```
And finally
```
doSomething(); //do the thing
``` 
no, I would not be happy with this. 
```
doSomething(); // do the thing
```
is what I want to see.

**Reply by sathishkumar2010** [reply]
yes these are okay i am trying to see my GitHub action failure that's the reason my code is not correct format.  But my action was success see my screen shot looks pass all checks.
